lib/Transformer.js

```diff
@@ -62,15 +63,34 @@ class Transformer extends Transform {
     const outputValues = this.transformers
       .map(fn => fn(sourceRow, this.info))
 
-    if (!this.filterFn(outputValues)) {
+    const filterFns = Array.isArray(this.filterFn) ? this.filterFn : [this.filterFn]
+
+    let filter = true
+
+    filterFns.forEach(fn => {
+      if (!fn(outputValues)) filter = false
+    })
+
+    if (!filter) {
       this.filterRowCount()
       return callback(null, null)
     }
 
-    this.transformerFn(
-      outputValues,
-      (err, transformedValues) => callback(err, csvEncoder(transformedValues))
-    )
+    const transformFns = Array.isArray(this.transformerFn) ? this.transformerFn : [this.transformerFn]
+
+    let transformedValues = outputValues
+
+    transformFns.forEach(fn => {
+      return new Promise((resolve, reject) => {
+        fn(transformedValues, (err, values) => {
+          if (err) return reject(err)
+          transformedValues = values
+          resolve()
+        })
+      })
+    })
+
```

Comment:
I am pretty confident that if this is working, then it's working by accident. I am not sure what the correct solution is, but let me have think.

Reply:
We need to pass the values through each transformer in turn, and once we've done all that ping it back out through `callback`. Our transfomer function interface isn't great, but we can work with it. Something like this should do it, I think
```
const transformerCascade = (err, values, index = 0) => {
  if (index !== transformFns.length && !err) {
     transformFns[index](values, (e, transformedValues) => transformerCascade(e, transformedValues, index+1))
  } else {
    callback(err, csvEncoder(values))
   }
 }

transformerCascade(null, outputValues)
```

Reply:
Actually we could wrap that up a bit and push it up the initialisation section too - 
```
this.transformers = (outputValues, cb) => {
  const transformerCascade = (err, values, index = 0) => {
    if (index !== transformFns.length && !err) {
       transformFns[index](values, (e, transformedValues) => transformerCascade(e, transformedValues, index+1))
    } else {
      cb(err, values)
    }
  }

  transformerCascade(null, outputValues)
}
```
then the call inside `_transform` collapses to 
```
this.transformers(
  outputValues,
  (cb, transformedValues) => callback(err, csvEncoder(transformedValues))
)
```

Reply:
Thank you!

Reply:
Thank you!